### PR TITLE
main/python3: upgrade to 3.7.1

### PR DESCRIPTION
**Pull request by viktorkholod**

### CODE REVIEW

**Comment by andypost** [comment]
3.7.1 is out https://www.python.org/downloads/release/python-371/

Also commits needs rebase

**Comment by andypost** [comment]
Just few tests needs fix
```
4 tests failed:
    test_c_locale_coercion test_embed test_ssl test_urllib2net
```

**Comment by viktorkholod** [comment]
@andypost, there is no way to fix `test_ssl` with current Travis-ci config, details are here https://bugs.python.org/issue32549

**Comment by andypost** [comment]
@viktorkholod the issue is not in travis but in Alpine's openssl library which in progress of transition to openssl 1.1 from libressl

File: main/python3/APKBUILD [hunk 1]
```diff
@@ -27,7 +27,6 @@ prepare() {
 	cd "$builddir"
 	# force system libs
 	rm -r Modules/expat \
-		Modules/zlib \
```
Comment (andypost) [inline main/python3/APKBUILD]: why that now needed? I mean it was removed but now we gonna compile it?

**Comment by andypost** [comment]
Also another question is how to test all py* packages that they works fine with 3.7

**Comment by andypost** [comment]
Probably it needs `options="net"`

File: main/python3/APKBUILD (hunk 1 above)
Comment (viktorkholod) [reply main/python3/APKBUILD]: zlib was removed from the Python source https://bugs.python.org/issue31358

Comment (viktorkholod) [reply main/python3/APKBUILD]: zlib was removed from the Python source https://github.com/python/cpython/pull/3375

**Comment by andypost** [comment]
@viktorkholod  please rebase commit on current master

**Comment by haizaar** [comment]
Can someone please update what's the status of Python3.7 in alpine? I see that this PR was closed without merging.

**Comment by maxwell-k** [comment]
I think https://bugs.alpinelinux.org/issues/9633 is the relevant feature request in the bug tracker

**Comment by haizaar** [comment]
Yes, I opened that one, and was referred here :)
Still wonder if anyone can provide more details / ETA.

**Pull request closed without merging**